public/app/features/explore/state/time.ts

```diff
@@ -160,6 +168,39 @@ export function zoomOut(scale: number): ThunkResult<void> {
   });
 }
 
+export function copyTimeRangeToClipboard(): ThunkResult<void> {
+  return (dispatch, getState) => {
+    const range = getState().explore.panes[Object.keys(getState().explore.panes)[0]]!.range.raw;
+    navigator.clipboard.writeText(JSON.stringify(range));
+
+    appEvents.emit(AppEvents.alertSuccess, ['Time range copied to clipboard']);
+  };
+}
+
+export function pasteTimeRangeFromClipboard(): ThunkResult<void> {
+  return async (dispatch, getState) => {
+    const raw = await navigator.clipboard.readText();
+    let range;
+
+    try {
+      range = JSON.parse(raw);
+    } catch (e) {
+      appEvents.emit(AppEvents.alertError, ['Invalid time range', `"${range}" is not a valid time range`]);
+      return;
+    }
+
+    const panesSynced = getState().explore.syncedTimes;
+
+    if (panesSynced) {
+      dispatch(updateTimeRange({ exploreId: Object.keys(getState().explore.panes)[0], rawRange: range }));
+      dispatch(updateTimeRange({ exploreId: Object.keys(getState().explore.panes)[1], rawRange: range }));
+      return;
+    }
+
+    dispatch(updateTimeRange({ exploreId: Object.keys(getState().explore.panes)[0], rawRange: range }));
```

Comment:
Currently updating only the left pane in split mode if panes are not synced. Maybe we can add some type of notification that tells user that, or add something to Explore docs.